File: LeetCode_26_removeDuplicatesFromSortedArray.py

from typing import List


class LeetCode_26_removeDuplicatesFromSortedArray:
    def removeDuplicates(self, nums: List[int]) -> int:
        slow = 0
        fast = 1
        while(fast < len(nums)):
            if(nums[slow] != nums[fast]):
                slow = slow + 1
                nums[slow] = nums[fast]
            else:
                fast = fast+1
        return slow + 1

File: LeetCode_26_removeDuplicatesFromSortedArray.java
class LeetCode_26_removeDuplicatesFromSortedArray {
    public int removeDuplicates(int[] nums) {
        
        if(nums.length == 1){
            return 1;
        }
        int pointer1 = 0, pointer2 = 1;
        while(pointer2 < nums.length){
            if(nums[pointer1] == nums[pointer2]){
                // ++pointer1;
                ++pointer2;
            }

            else{
                ++pointer1;
                nums[pointer1] = nums[pointer2];
            }
        }
        return pointer1 + 1;

    }
}